pkg/scheduler/worker/workerpool.go

```diff
@@ -238,3 +244,33 @@ func (w *Pool) RunningWorkers() (int, error) {
 func (w *Pool) Size() int {
 	return w.config.PoolSize
 }
+
+func (w *Pool) RegisterObserver(observer occupancy.Observer) {
+	if w.occupancyObserver != nil {
+		w.logger.Debug("Worker pool is already registered an occupancy observer")
+		return
+	}
+	w.occupancyObserver = observer
+	w.logger.Infof("Worker pool registered new occupancy observer: %v", observer)
+}
+
+func (w *Pool) UnregisterObserver(observer occupancy.Observer) {
+	if w.occupancyObserver == nil {
+		w.logger.Debug("Worker pool has nil occupancy observer already")
+		return
+	}
+	w.occupancyObserver = nil
+	w.logger.Infof("Worker pool unregistered its occupancy observer: %v", observer)
+}
+
+func (w *Pool) Notify() error {
+	if w.occupancyObserver == nil {
+		return fmt.Errorf("worker pool failed to notify occupancy observer: %v", w.occupancyObserver)
```

Comment:
This Error message is a bit misleading, since the error is not that the notify failed but instead that no observer is registered

Reply:
Thanks! I have changed the error message to `no observer was registered`